app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,68 @@
+require 'date'
+
+class TasksController < ApplicationController
+  def index
+    @tasks = Task.all
+  end
+
+  def show
+    task_id = params[:id].to_i
+    @task = Task.find(task_id)
+  end
+
+  def new
+    @task = Task.new
+  end
+
+  def create
+    @task = Task.new(action: params[:task][:action],
+                    description: params[:task][:description])
+    if @task.save
+      redirect_to root_path
+    else
+      render :new
+    end
+  end
+
+  def edit
+    @task = Task.find_by(id: params[:id])
+  end
+
+  def update
+    @task = Task.find_by(id: params[:id])
+
+    if @task.update(task_params)
+      redirect_to task_path(@task.id)
+    else
+      render :update
+    end
+  end
+
+  def destroy
+    task = Task.find_by(id: params[:id])
+
+    task.destroy
+    redirect_to root_path
+  end
+
+  def complete
+    task = Task.find(params[:id])
+    task.completion_date = DateTime.now
```

Comment:
There aren't ways right now that we can observe "what happens if we try to complete a Task that doesn't exist," but keep in mind... if `Task.find(params[:id])` doesn't find a Task, what will happen?